haystack/document_stores/filter_utils.py

```diff
@@ -1,6 +1,12 @@
 from typing import Union, List, Dict
 from abc import ABC, abstractmethod
 from collections import defaultdict
+from functools import reduce
+
+from haystack.utils.import_utils import safe_import
```

Comment:
I can't see where `safe_import` is used.

Reply:
You are right, this import is unnecessary. I've removed it now.